### PR TITLE
Expanding the Vue tutorial and fixing typo from data-model.md page

### PR DESCRIPTION
**Pull request by dbhagen**
*Description of changes:*

Expanding the Vue tutorial with additional explanation and block sections for `NPM` vs. `Yarn`.

This is my first pass, just checking in my work so far so I can get feedback on the direction I'm going. I have more pages to work through.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by swaminator** [comment]
@dbhagen thanks for the PR. One concern is with the # descriptions in the CLI flow, the content will get too long and not look good. mind sharing a screenshot of what it looks like?

**Comment by mauerbac** [comment]
Hey @swaminator and @dbhagen - I just took a look and ran the docs locally. I attached a few screenshots. IMO - it always helps to have clearer instructions and I don't think it makes the tutorial too long. My only thought was React and other libs don't have this level of detail, so does it create an imbalance. Since this is an improvement I'm good with merging it. Since Nikhil is out, thoughts @renebrandel ? 
Screen Shot 2020-07-17 at 1 21 07 PM
Screen Shot 2020-07-17 at 1 21 13 PM

**Comment by dbhagen** [comment]
My intention was to go through the React Code next and go for consistency. Unfortunately, work got crazy and and I haven’t had a chance to work on that. But if this seems like a doable direction, I’ll knock out the react version.

File: docs/start/getting-started/fragments/vue/setup.md [hunk 1]
```diff
@@ -25,33 +69,139 @@ amplify init
...
 
-# Sometimes the CLI will prompt you to edit a file, it will use this editor to open those files.
-Choose your default editor
+Set the backend environment name to be `dev`
+```console
+? Enter a name for the environment dev
+```
 
+Sometimes the CLI will prompt you to edit a file, it will use this editor to open those files.
+Select your prefered code editor software
```
Comment (ericclemmons) [inline docs/start/getting-started/fragments/vue/setup.md:87]: Spell check failed in CI.

```suggestion
Select your preferred code editor software
```

**Pull request merged**